src/GitHub.App/Models/PullRequestModel.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Globalization;
+using GitHub.Primitives;
+using GitHub.VisualStudio.Helpers;
+using NullGuard;
+
+namespace GitHub.Models
+{
+    public sealed class PullRequestModel : NotificationAwareObject, IPullRequestModel
+    {
+        public PullRequestModel(int number, string title, IAccount author, DateTimeOffset createdAt, DateTimeOffset? updatedAt = null)
+        {
+            Number = number;
+            Title = title;
+            Author = author;
+            CreatedAt = createdAt;
+            UpdatedAt = updatedAt ?? CreatedAt;
+        }
+
+        public void CopyFrom(IPullRequestModel other)
+        {
+            if (Number != other.Number)
+                throw new ArgumentException("Instance to copy from must match", nameof(other));
```

Comment:
Probably would be useful to print out what the current ID is and what the other one was.